ibm/mas_devops/roles/cp4d_wds/tasks/embeddedwds.yml

```diff
@@ -0,0 +1,310 @@
+---
+# 1. Check if CP4D is installed and fail if not there
+# -----------------------------------------------------------------------------
+- name: "cp4d/wds: Check if cp4d instance is there"
+  community.kubernetes.k8s_info:
+    api_version: cpd.ibm.com/v1
+    kind: Ibmcpd
+  register: cp4dalllist
+
+- name: "Fail if no cp4d instance is provided before discovery install"
+  when: cp4dalllist.resources| length == 0
+  fail:
+    msg: "CP4D Base is required to WDS install and instance provision"
+
+# 2. Query the existing CP4D used project, Storage Class
+# -----------------------------------------------------------------------------
+- name: "cp4d/wds: cp4d cluster status and query the project and Storage Class used by CP4D"
+  when: cp4dalllist.resources| length != 0
+  set_fact:
+    cpdavailable: '{% if cp4dalllist.resources[0].status.controlPlaneStatus == "Completed" %}True{% else %}False{% endif %}'
+    cpdnamespace: "{{ cp4dalllist.resources[0].metadata.namespace }}"
+    cpdstorageclass: "{{ cp4dalllist.resources[0].spec.zenCoreMetadbStorageClass is defined | ternary(cp4dalllist.resources[0].spec.zenCoreMetadbStorageClass,wdsstorageclass) }}"
+
+- name: "cp4d/wds: Print the storage class used by discovery install"
+  debug:
+    msg:
+      - "Project for Discovery install .......... {{ cpdnamespace }}"
+      - "Storage Class used by Discovery ........ {{ cpdstorageclass }}"
+
+# 3. Query the existing CP4D admin login crdential
+# -----------------------------------------------------------------------------
+
+- name: "cp4d/wds: check if the cp4d admin user secret exist"
+  when: cpdavailable is defined and cpdavailable
+  community.kubernetes.k8s_info:
+    api_version: v1
+    kind: Secret
+    name: "admin-user-details"
+    namespace: "{{ cpdnamespace }}"
+  register: cpdadmincred
+
+- name: "cp4d/wds: Query the existing cp4d instance Login Credential"
+  when:
+    - cpdavailable is defined and cpdavailable
+    - cpdadmincred.resources| length != 0
+  set_fact:
+    wds_admin_username: "admin"
+    wds_admin_password: "{{ cpdadmincred.resources[0]['data']['initial_admin_password'] | b64decode }}"
```

Comment:
Could the user have changed this password so that initial_admin_password is no longer correct?

Reply:
Good point. I add logic to permit user to provide admin user/password if he already changed the initial password.